api/types/role.go

```diff
@@ -1829,7 +1829,7 @@ func (r *RoleV6) GetLabelMatchers(rct RoleConditionType, kind string) (LabelMatc
 		return LabelMatchers{cond.NodeLabels, cond.NodeLabelsExpression}, nil
 	case KindKubernetesCluster:
 		return LabelMatchers{cond.KubernetesLabels, cond.KubernetesLabelsExpression}, nil
-	case KindApp:
+	case KindApp, KindSAMLIdPServiceProvider:
```

Comment:
Can you add a comment here and below on why we're applying app_labels matches to SAML SP resources as well.

Reply:
Comment added in https://github.com/gravitational/teleport/pull/44113/commits/dd68cf346926120ee7fa62df88507e54cc38d6eb. 

Noted on the backport and changelog. And I will update changelog once all the foloowup PRs are n.